launcher/launcher.go

```diff
@@ -1186,6 +1186,18 @@ func (launcher *Launcher) startInstance(instance *instanceInfo) error {
 		instance.setRunStatus(runStatus)
 	}
 
+	if err := launcher.instanceMonitor.StartInstanceMonitor(
```

Comment:
You moved Start monitoring under lock, and serialized this call. Lock guards instance.runStatus as it may be changed in another thread. So, remove unlock from defer and put it after checking run status condition.

Reply:
fixed